src/services/poolTogether.js

```diff
@@ -0,0 +1,416 @@
+import POOL_DAI_ABI from 'abi/poolDAI.json';
+import POOL_USDC_ABI from 'abi/poolUSDC.json';
+import DAI_ABI from 'abi/DAI.json';
+import USDC_ABI from 'abi/USDC.json';
+
+import smartWalletService from './smartWallet';
+
+const POOL_TOGETHER_NETWORK = NETWORK_PROVIDER === 'ropsten' ? 'kovan' : NETWORK_PROVIDER;
+
+const getPoolTogetherTokenContract = (symbol: string) => {
+  const poolContractAddress = symbol === DAI ? POOL_DAI_CONTRACT_ADDRESS : POOL_USDC_CONTRACT_ADDRESS;
+  const poolAbi = symbol === DAI ? POOL_DAI_ABI : POOL_USDC_ABI;
+  const unitType = symbol === DAI ? 18 : 6; // DAI 18 decimals, USDC 6 decimals
+  const provider = getEthereumProvider(POOL_TOGETHER_NETWORK);
+  const poolContract = new Contract(poolContractAddress, poolAbi, provider);
+
+  const tokenContractAddress = symbol === DAI ? DAI_ADDRESS : USDC_ADDRESS;
+  const tokenABI = symbol === DAI ? DAI_ABI : USDC_ABI;
+  const tokenContract = new Contract(tokenContractAddress, tokenABI, provider);
+
+
+  return {
+    poolContract,
+    poolContractAddress,
+    poolAbi,
+    unitType,
+    provider,
+    tokenContractAddress,
+    tokenABI,
+    tokenContract,
+  };
+};
+
+export async function getPoolTogetherInfo(symbol: string, address: string): Promise<PoolInfo> {
+  const {
+    poolContract: contract,
+    unitType,
+    provider,
+    poolContractAddress: contractAddress,
+  } = getPoolTogetherTokenContract(symbol);
+  const accountedBalance = await contract.accountedBalance();
+  const balanceCallData = contract.interface.functions.balance.encode([]);
+  const result = await provider.call({ to: contract.address, data: balanceCallData });
+  const balance = contract.interface.functions.balance.decode(result);
+  const balanceTakenAt = new Date();
+  const currentOpenDrawId = await contract.currentOpenDrawId();
+  const currentDraw = await contract.getDraw(currentOpenDrawId);
+  const committedSupply = await contract.committedSupply();
+  const openSupply = await contract.openSupply();
+
+  let userInfo;
+  try {
+    const totalBalance = await contract.totalBalanceOf(address); // balance including open, committed, fees, sponsorship
+    if (!totalBalance.eq(0)) {
+      const openBalance = await contract.openBalanceOf(address); // balance in the open Draw
+      const userCurrentPoolBalance = await contract.balanceOf(address); // balance in the current committed Draw
+      const ticketBalance = openBalance.add(userCurrentPoolBalance); // this is the current ticket balance in total
+      userInfo = {
+        openBalance: formatMoney(utils.formatUnits(openBalance.toString(), unitType)),
+        totalBalance: formatMoney(utils.formatUnits(totalBalance.toString(), unitType)),
+        userCurrentPoolBalance: formatMoney(utils.formatUnits(userCurrentPoolBalance.toString(), unitType)),
+        ticketBalance: formatMoney(utils.formatUnits(ticketBalance.toString(), unitType)),
+      };
+    }
+  } catch (e) {
+    reportLog('Error checking PoolTogether User Info', {
```

Comment:
Is something supposed to be returned here in the event that total balance check errors?

Reply:
Thanks, added a failsafe and error report to all calls not to update the state/storage if an error is thrown it will return null, 4bc3f3e

Reply:
👍  thank you